plugins/inventory/lxd.py

```diff
@@ -0,0 +1,1056 @@
+__metaclass__ = type
+
+DOCUMENTATION = r'''
+    name: community.general.lxd
+    short_description: Returns Ansible inventory from lxd host
+    description:
+        - Get inventory from the lxd.
+        - Uses a YAML configuration file that ends with 'lxd.(yml|yaml)'.
+    version_added: "2.5.0"
+    author: "Frank Dornheim (@conloos)"
+    options:
+        plugin:
+            description: Token that ensures this is a source file for the 'lxd' plugin.
+            required: true
+            choices: [ 'community.general.lxd' ]
+        url:
+            description:
+            - The unix domain socket path or the https URL for the lxd server.
+            - Sockets in filesystem have to start with C(unix:).
+            - Mostly C(unix:/var/lib/lxd/unix.socket) or C(unix:/var/snap/lxd/common/lxd/unix.socket).
+            default: unix:/var/snap/lxd/common/lxd/unix.socket
+            required: false
+            type: str
+        client_key:
+            description:
+            - The client certificate key file path.
+            required: false
+            aliases: [ key_file ]
+            default: $HOME/.config/lxc/client.key
+            type: path
+        client_cert:
+            description:
+            - The client certificate file path.
+            required: false
+            aliases: [ cert_file ]
+            default: $HOME/.config/lxc/client.crt
+            type: path
+        trust_password:
+            description:
+            - The client trusted password.
+            - You need to set this password on the lxd server before
+                running this module using the following command
+                C(lxc config set core.trust_password <some random password>)
+                See U(https://www.stgraber.org/2016/04/18/lxd-api-direct-interaction/).
+            - If I(trust_password) is set, this module send a request for authentication before sending any requests.
+            required: false
+            type: str
+        state:
+            description: Filter the container according to the current status.
+            required: false
+            type: str
+            default: none
+            choices: [ 'STOPPED', 'STARTING', 'RUNNING', 'none' ]
+        prefered_container_network_interface:
+            description:
+            - If a container has multiple network interfaces, select which one is the prefered as pattern.
+            - Combined with the first number that can be found e.g. 'eth' + 0.
+            required: false
+            type: str
+            default: eth
+        prefered_container_network_family:
+            description:
+            - If a container has multiple network interfaces, which one is the prefered by family.
+            - Specify C(inet) for IPv4 and C(inet6) for IPv6.
+            required: false
+            type: str
+            default: inet
+            choices: [ 'inet', 'inet6' ]
+        dispose:
```

Comment:
In all APIs I know, `dispose` means "throw something away". Why not use something more explicit which contains `group(s)`, like `groups_from_values`?

Reply:
I agree that `dispose` is possibly not the best term to use here. I think the terminology `groupby` kind of fits in this case, what do you guys think?

Reply:
changed to groupby